### PR TITLE
feat(react-calendar-compat): changed design for the selected state

### PR DESCRIPTION
**Pull request by ValentinaKozlova**
- Made changes for selected state according to the latest design - #30426
- Made borders-radius `4px` for consistency
## 
| Feature| Previous Behavior | New Behavior | 
| ----- | -------- | -------- | 
**Selected state:**
| Day|image | image|
| Day today|image | image|
| Focus state|image | image
| Focus state - day today|image |image
| **border-radius changes:** | none,  2px | 4px |
| Hover state - day|image | image
| Hover state - month|image | image
| Hover state - year|image |image
| Hover state - arrow|image |image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-calendar-compat</samp> <br /> <abbr title='bundle-size/Calendar.fixture.js'>Calendar Compat</abbr> | `146.465 kB`<br />`39.03 kB` | `150.361 kB`<br />`40.088 kB` | `3.896 kB` <br />`1.058 kB` |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr> | `220.038 kB`<br />`62.373 kB` | `223.932 kB`<br />`63.411 kB` | `3.894 kB` <br />`1.038 kB` |

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/dda502daecc94c51c1148c403a4c386a2317775a'>dda502daecc94c51c1148c403a4c386a2317775a</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-calendar-compat-acdf3646-9c24-46d9-85eb-7f68e84c551d.json
Comment (fabricteam) [inline change/@fluentui-react-calendar-compat-acdf3646-9c24-46d9-85eb-7f68e84c551d.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31509&commitId=a839c3c03a1d1dc9f087a1e0cb58fde0c97c2eac&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|2| Changed|

</details>


  <details><summary><b>Calendar Compat</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Calendar Compat.Default.default.chromium.png|1612| Changed|
|Calendar Compat.Custom icons.default.chromium.png|1612| Changed|
|Calendar Compat.Default - RTL.default.chromium.png|1612| Changed|

</details>


  <details><summary><b>DatePicker Compat</b> 16 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|DatePicker Compat.DateRange- work week.default.chromium.png|1680| Changed|
|DatePicker Compat.DateRange- month - Dark Mode.default.chromium.png|2510| Changed|
|DatePicker Compat.DateRange- day.default.chromium.png|1612| Changed|
|DatePicker Compat.DateRange- month.default.chromium.png|1593| Changed|
|DatePicker Compat.marked dates.default.chromium.png|1626| Changed|
|DatePicker Compat.default - Dark Mode.opened.chromium.png|1665| Changed|
|DatePicker Compat.default.opened.chromium.png|1612| Changed|
|DatePicker Compat.showWeekNumbers.default.chromium.png|1743| Changed|
|DatePicker Compat.DateRange- week.default.chromium.png|1722| Changed|
|DatePicker Compat.DateRange- month - High Contrast.default.chromium.png|2494| Changed|
|DatePicker Compat.marked dates - High Contrast.default.chromium.png|2074| Changed|
|DatePicker Compat.showWeekNumbers - Dark Mode.default.chromium.png|1767| Changed|
|DatePicker Compat.default - High Contrast.opened.chromium.png|2053| Changed|
|DatePicker Compat.showWeekNumbers - High Contrast.default.chromium.png|2290| Changed|
|DatePicker Compat.marked dates - Dark Mode.default.chromium.png|1687| Changed|
|DatePicker Compat.when rendering inline, it should not render behind relatively positioned elements.default.chromium.png|1612| Changed|

</details>


  <details><summary><b>SwatchPicker Converged</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.Shape.default.chromium.png|0| Removed|
|SwatchPicker Converged.default - RTL.default.chromium.png|0| Removed|
|SwatchPicker Converged.default.default.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Comment (sopranopillow) [reply change/@fluentui-react-calendar-compat-acdf3646-9c24-46d9-85eb-7f68e84c551d.json:1]: Calendar changes look good to me, the others I think are from previous PRs(?)

**Review by sopranopillow — DISMISSED** [review]
Approving with the following work item pending #31551

Thanks for the changes!

File: packages/react-components/react-calendar-compat/src/components/CalendarDayGrid/useCalendarDayGridStyles.styles.ts
Comment (sopranopillow) [inline packages/react-components/react-calendar-compat/src/components/CalendarDayGrid/useCalendarDayGridStyles.styles.ts:458]: We usually add the classname first and then the other styles. this classname is mainly used for selectors rather than styling.

```suggestion
    dayTodayMarker: mergeClasses(calendarDayGridClassNames.dayTodayMarker, dayTodayMarkerStyles.base),
```

**Review by sopranopillow — APPROVED** [review]
Some changes that should go in before merging, sorry to catch them later

File: change/@fluentui-react-calendar-compat-acdf3646-9c24-46d9-85eb-7f68e84c551d.json
Comment (ValentinaKozlova) [reply change/@fluentui-react-calendar-compat-acdf3646-9c24-46d9-85eb-7f68e84c551d.json:1]: I think, yes. didn't change anything else. DatePicker has expected changes as it uses CalendarPicker

**Pull request merged**
